x-pack/plugins/ml/server/models/data_frame_analytics/models_provider.ts

```diff
@@ -40,11 +35,38 @@ interface TrainedModelStatsResponse extends MlTrainedModelStats {
   model_size_stats?: TrainedModelModelSizeStats;
 }
 
-export function modelsProvider(
-  client: IScopedClusterClient,
-  mlClient: MlClient,
-  memoryOverviewService?: MemoryOverviewService
-) {
+export interface MemoryStatsResponse {
```

Comment:
there is a lot of cross over between types in this file and the common `NodeDeploymentStatsResponse` interface.
Would it be possible to combine them to remove the duplication? 
Possibly by breaking up `NodeDeploymentStatsResponse` into component parts that can be reused.

Reply:
I've deliberately kept `NodeDeploymentStatsResponse`. It provides modes more detailed memory breakdown (by model for trained models) and potentially will be extended for other ML entities.

Reply:
i see, so this type is only here for the `mlClient.getMemoryStats` call. as there is no proper type in the client yet.
Could you please add a quick comment to this line? To explain that it should be temporary while elastic/elasticsearch-specification#1538 exists.